src/core/integrate.cpp

```diff
@@ -407,24 +460,26 @@ int integrate(int n_steps, int reuse_forces) {
         fluid_step += 1;
         if (fluid_step >= lb_steps_per_md_step) {
           fluid_step = 0;
-          LB::propagate();
-          EK::propagate();
+          lb.propagate();
+          ek.propagate();
         }
         lb_lbcoupling_propagate();
       } else if (lb_active) {
-        auto const lb_steps_per_md_step = LB::get_steps_per_md_step(time_step);
+        auto &lb = system.lb;
+        auto const lb_steps_per_md_step = lb.get_steps_per_md_step(time_step);
```

Comment:
Good catch! Not only that, but the assertion was also incorrect, because the EK propagation counter was not synchronized with the LB propagation counter. Let me fix that.

Reply:
By the way, we should really think about writing a test for tau > dt with LB + EK active.